src/db.c

```diff
@@ -2503,6 +2514,11 @@ keyStatus expireIfNeeded(redisDb *db, robj *key, kvobj *kv, int flags) {
         if (!(flags & EXPIRE_FORCE_DELETE_EXPIRED)) return KEY_EXPIRED;
     }
 
+    /* Check if user configuration disables lazy-expire deletions in current state.
+       This will only apply if the server doesn't mandate key deletion to operate correctly (write commands). */
```

Comment:
```suggestion
    /* Check if user configuration disables lazy-expire deletions in current state.
     * This will only apply if the server doesn't mandate key deletion to operate correctly (write commands). */
```